src/LinearRemapFV.cpp

```diff
@@ -1475,32 +1475,47 @@ void LinearRemapFVtoGLL(
 
 			const NodeVector & nodesOverlap = meshOverlap.nodes;
 
-			int nOverlapTriangles = faceOverlap.edges.size() - 2;
-
 			// Quantities from the Second Mesh
 			int ixSecond = meshOverlap.vecTargetFaceIx[ixOverlap + i];
 
 			const NodeVector & nodesSecond = meshOutput.nodes;
 
 			const Face & faceSecond = meshOutput.faces[ixSecond];
-
+			int nbEdges = faceOverlap.edges.size();
+            int nOverlapTriangles = 1;
+            Node center; // not used if nbEdges == 3
+            if (nbEdges > 3) { // decompose from center in this case
+                nOverlapTriangles = nbEdges;
+                for (int k = 0; k < nbEdges; k++) {
+                    const Node &node = nodesOverlap[faceOverlap[k]];
+                    center = center + node;
+                }
+                center = center / nbEdges;
+                double magni = sqrt(
+                        center.x * center.x + center.y * center.y
+                                + center.z * center.z);
+                center = center / magni; // project back on sphere of radius 1
+            }
+
+            Node node0, node1, node2;
+            double dTriArea;
 			// Loop over all sub-triangles of this Overlap Face
 			for (int j = 0; j < nOverlapTriangles; j++) {
 
-				// Cornerpoints of triangle
-				const Node & node0 = nodesOverlap[faceOverlap[0]];
-				const Node & node1 = nodesOverlap[faceOverlap[j+1]];
-				const Node & node2 = nodesOverlap[faceOverlap[j+2]];
-
-				// Calculate the area of the modified Face
-				Face faceTri(3);
-				faceTri.SetNode(0, faceOverlap[0]);
-				faceTri.SetNode(1, faceOverlap[j+1]);
-				faceTri.SetNode(2, faceOverlap[j+2]);
-
-				double dTriArea =
-					CalculateFaceArea(faceTri, nodesOverlap);
-
+			    if (nbEdges == 3) // will come here only once, nOverlapTriangles == 1 in this case
+                {
+                    node0 = nodesOverlap[faceOverlap[0]];
+                    node1 = nodesOverlap[faceOverlap[1]];
+                    node2 = nodesOverlap[faceOverlap[2]];
+                }
+                else // decompose polygon in triangles around the center
+                {
```

Comment:
Good opportunity to take out all of this repeated code for polygon decomposition and refactor to a single method ?

Reply:
I don't think it is worth adding another indirection; decomposition is classic, and it has to be skipped for pure triangles (it does not make sense to decompose a triangle in 3 more triangles). But for a polygon with at least 4 sides, the decomposition needs to be consistent, irrespective where the first node is in connectivity